src/main/java/io/quarkus/bot/TriageIssue.java

```diff
@@ -84,6 +84,26 @@ void triageIssue(@Issue.Opened GHEventPayload.Issue issuePayload,
                 LOG.info("Issue #" + issue.getNumber() + " - Add label: " + Labels.TRIAGE_NEEDS_TRIAGE);
             }
         }
+//
+//        if(GHIssues.hasAreaLabel(issue) && !triaged){
+//            if  (!quarkusBotConfig.isDryRun())  {
+//                issue.comment("/cc @" + String.join(", @", mentions));
+//            } else {
+//                LOG.info("Issue #" + issue.getNumber() + " - Mentions: " + String.join(", ", mentions));
+//            }
+//        }
+
+        if (labels.contains(Labels.TRIAGE_NEEDS_TRIAGE) && GHIssues.hasAreaLabel(issue)) {
+            if (labels.remove(Labels.TRIAGE_NEEDS_TRIAGE)) {
+                if (!mentions.isEmpty()) {
+                    if (!quarkusBotConfig.isDryRun()) {
+                        issue.comment("/cc @" + String.join(", @", mentions));
+                    } else {
+                        LOG.info("Issue #" + issue.getNumber() + " - Mentions: " + String.join(", ", mentions));
+                    }
+                }
+            }
+        }
```

Comment:
Hmmm, how do you think this will work? Because it's not doing what it should do at all.

You're breaking a perfectly valid rule and you don't react to the right event. This should be a new rule. Please read the description of the workflow in the issue. You shouldn't react to the fact that an issue is opened but to the fact that the `triage/needs-triage` label is removed.

Reply:
If the issue is open and label `triage/needs-triage` is marked along with some `area/..` labels that we added manually then the above actions should take place. This is what wanted no?

Reply:
>You shouldn't react to the fact that an issue is opened but to the fact that the `triage/needs-triage` label is removed.

oh alright, perhaps I misunderstood it! let me try :)

Reply:
@gsmet 
One more thing I wanted to ask, how do we test the bot? I was able to test the "personal testing bot" very well but if I run `mvn quarkus:dev` for `quarkus-bot` it asks for app-id and all those params that I added in my `.env' file.

Reply:
@SaumyaSingh1 I think the important thing to have in mind is that from the start the issue has been created already.

The rule you modified shouldn't be modified. When your rule arrives, it has already been applied and the bot wasn't able to triage things automatically.

The current situation is the following:
1. I have an issue the bot hasn't triaged as it didn't match any rule which has the triage/needs-rebase label
2. I manually add some labels depending on the content of the issue
3. I remove the needs-rebase label
4. I manually ping people (and I often have to go read the config file to find the right people)

What I want is to get rid of 4. . So what you need is react to 3. by listening to the `@Issue.Unlabeled` event and check if the removed label is `triage/needs-rebase`.

But now that I write that, I think you will miss an information in the GitHub API payload, unfortunately. It looks like you won't have the removed label in the payload. I will work on adding that in GitHub API and get back to you when I have a release including this piece of data.

I think for now you should start with https://github.com/quarkusio/quarkus-bot/issues/30 . It should be simple to implement and will get you started.

What I suggest you is to think about it a bit then post in the issue how you would implement it (typically the event you would listen for instance). And we can continue from there.

As for quarkus-bot, we don't provide a `.env` file in the repo (it contains a private key so we cannot do that) so you need to add your own and link it to the GitHub App you already created and it should work OK. You can reuse the one you already created for your personal testing bot.

Reply:
I created https://github.com/hub4j/github-api/issues/1060 to track the GH API issue.

Reply:
Thanks a lot for all the explanation @gsmet!

Now the procedure is much clear to me. Initially, I misunderstood the issue description and the whole process. I have made the PR for #30 :)

Reply:
Hey @SaumyaSingh1 ,

I updated the GitHub API and we now have the missing field. You will need to rebase to get it (e.g. something like `git pull --rebase upstream main`)

The idea here would be to:
- create a specific class `PingWhenNeedsTriageRemoved`
- listen to the `@Issue.Unlabeled` event
- in the `GHEventPayload.Issue` you will get, the `getLabel()` method will now return the removed label so you can check if it's the `needs-triage` label and if it is, you can try to apply the rules to ping the right persons.

Don't hesitate to ping me if things are unclear :).

Reply:
Updated @gsmet :)